bentoml/frameworks/tensorflow.py

```diff
@@ -246,23 +240,39 @@ def pack(
                         "version 1",
                         str(options),
                     )
-
                 tf.saved_model.save(
                     obj, self._tmpdir.name, signatures=signatures,
                 )
             self._path = self._tmpdir.name
         else:
             self._path = obj
-
+        self._packed = True
+        loaded = self.get()
+        logger.warning(
+            f"Due to TensorFlow's save and load mechanism, only methods marked by "
+            f"`tf.function` (and some keras defaults) could be restored after "
+            f"packing & loading.\n"
+            f"You can test the restored model object by referring:\n"
+            f"<bento_svc>.artifacts.{self.name}\n"
+        )
+        logger.info(pretty_format_restored_model(loaded))
+        if hasattr(loaded, "keras_api"):
+            logger.warning(
+                f"You may be using `{self.__class__.__name__}` to pack a keras model. "
```

Comment:
`{self.__class__.__name__}` is being used to pack a TF model built with Keras API. In order to get optimal serving performance, we recommend either replacing `{self.__class__.__name__}` with `KerasModelArtifact`, or wrapping the `keras_model.predict` method with `tf.function` decorator.